applications/GeoMechanicsApplication/custom_constitutive/thermal_filter_law.cpp

```diff
@@ -31,18 +31,14 @@ ConstitutiveLaw::Pointer GeoThermalFilterLaw::Clone() const
     return Kratos::make_shared<GeoThermalFilterLaw>(*this);
 }
 
-SizeType GeoThermalFilterLaw::WorkingSpaceDimension() { return mNumberOfDimensions; }
+SizeType GeoThermalFilterLaw::WorkingSpaceDimension() { return 1; }
 
-Matrix GeoThermalFilterLaw::CalculateThermalFilterMatrix(const Properties&  rProp,
-                                                                 const ProcessInfo& rProcessInfo) const
+Matrix GeoThermalFilterLaw::CalculateThermalFilterMatrix(const Properties& rProp, const ProcessInfo& rProcessInfo) const
 {
     KRATOS_TRY
 
-    Matrix result = ZeroMatrix(mNumberOfDimensions, mNumberOfDimensions);
-
-    const auto x = static_cast<int>(indexThermalFlux::X);
-    result(x, x) = rProp[THERMAL_CONDUCTIVITY_WATER];
-
+    Matrix result = ZeroMatrix(1, 1);
+    result(0, 0) = rProp[THERMAL_CONDUCTIVITY_WATER];
```

Comment:
There is no need to call ZeroMatrix because 0 will be over-written with rProp.... You may 
Matrix result(1,1); then 
result(0, 0) = rProp[THERMAL_CONDUCTIVITY_WATER];
or one line
auto result = ScalarMatrix(1,1,rProp[THERMAL_CONDUCTIVITY_WATER]);